File: 2545-sort-the-students-by-their-kth-score/2545-sort-the-students-by-their-kth-score.py
class Solution(object):
    def sortTheStudents(self, score, k):
        """
        :type score: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        sorted_dic = {}
        for r in range(len(score)):
            sorted_dic[score[r][k]] = r
        sorted_dic = sorted(sorted_dic.items(), reverse=True)
        new_score = []
        i=0
        for n in sorted_dic:
            new_score.append(score[n[1]])
            i += 1
        return new_score

File: 2545-sort-the-students-by-their-kth-score/2545-sort-the-students-by-their-kth-score.cpp
class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>>& score, int k) {
        map<int, int, greater<int>> kthscore;
        for (int r=0; r<score.size(); r++) 
            kthscore[score[r][k]] = r;
        
        vector<vector<int>> sorted_result;
        for (auto it=kthscore.begin(); it!=kthscore.end(); it++)
            sorted_result.push_back(score[it->second]);
        return sorted_result;
    }
};